src/hypothesis/internal/compat.py

```diff
@@ -530,6 +530,7 @@ def bad_django_TestCase(runner):
             from django.test import TransactionTestCase
             from hypothesis.extra.django import HypothesisTestCase
             _cases[:] = TransactionTestCase, HypothesisTestCase
-        except ImportError:
-            _cases[:] = type, type
+        except Exception:
```

Comment:
Hey @Zac-HD, I haven't thought deeply about this and I don't think I'm going to on my own, so I'll ask you: Is it possible there's an error here we really *should* show the user?

My guess is that the only case where the non-ImportError exception matters to use is if they really do want to use Django. And in that case, they'll see their error again later on when the code really tries to use Django.

So that would imply this is ok, and hiding the exception (which usually scares me a lot) is fine? But I'm not confident, so thought I'd ask.

Reply:
> My guess is that the only case where the non-ImportError exception matters to use is if they really do want to use Django. And in that case, they'll see their error again later on when the code really tries to use Django.

> So that would imply this is ok, and hiding the exception (which usually scares me a lot) is fine?

Yep, that's exactly right.  There are two reasons we need to catch `Exception` rather than a subtype:

1. If Django is installed in your environment but not used in this project, importing it will raise a configuration error (ie #1167)
2. We can't catch just that error, because importing the exception type would be enough to trigger it (!), and in any case there is no parent `DjangoException` type that users could catch to get everything (I really like our `HypothesisException` hierarchy)

We can be confident that this is safe, because if the user can't import Django they can't define a test using the Django runner without our extension.  Since preventing that is the whole purpose of this check, short-circuiting is fine in this case.

Reply:
Cool. If I were going to suggest anything different it might be `logging.exception(...)` (only for the non-ImportError case), since it's plausibly something the user's interested in knowing about.

But also I'm fine not doing that.

Reply:
With the sheer depth in the internals, and because this is designed for legibility rather than correctness, I'd rather not introduce logging to our codebase here!

Reply:
Cool, makes sense. Nice fix.